guides/plugins/plugins/framework/event/finding-events.md

```diff
@@ -0,0 +1,184 @@
+# Finding events
+
+## Overview
+
+Shopware 6 is fully extensible via plugins.
+Part of this extensibility is the usage of events, upon which one could react.
+
+This guide will cover how you can find those events in the first place, in order to use them in your plugin.
+
+## DAL Events
+
+At first we'll start with the [Data Abstraction Layer events](../data-handling/using-database-events.md).
+They're fired whenever a [DAL entity](../data-handling/add-custom-complex-data.md) is read, written, created, deleted, etc.
+
+There usually is no need to find them, since the pattern for them is always the same.
+You can use them by following this pattern: `entity_name.event`.
+For products, this could be e.g. `product.written` or `product.deleted`. For your custom entity, this then would be 
+`custom_entity.written` or `custom_entity.deleted`.
+
+However, some default Shopware entities come with special "Event classes", which are basically a class, which contains all
+possible kinds of events as constants.
+Have a look at the [product event class](https://github.com/shopware/platform/blob/v6.4.0.0/src/Core/Content/Product/ProductEvents.php) for example.
+This way you can also find out about all the possible DAL events available in Shopware.
+
+Finding those "event classes" can be done by searching for the term `@Event` in your project.
+
+You can use those events in a [subscriber](../../plugin-fundamentals/listening-to-events.md) like the following:
+```php
+public static function getSubscribedEvents(): array
+{
+    return [
+        ProductEvents::PRODUCT_LOADED_EVENT => 'onProductsLoaded',
+        'custom_entity.written' => 'onCustomEntityWritten'
+    ];
+}
+```
+
+As you can see, you can either use the event class constants, if available, or the string itself.
+
+You'll then have access to several event specific information, e.g. your listener method will have access to an [EntityWrittenEvent](https://github.com/shopware/platform/blob/v6.4.0.0/src/Core/Framework/DataAbstractionLayer/Event/EntityWrittenEvent.php)
+instance when subscribing to the `written` event.
+
+```php
+public function onCustomEntityWritten(EntityWrittenEvent $event): void
+{
+}
+```
+
+You can find all of those DAL event classes [here](https://github.com/shopware/platform/tree/v6.4.0.0/src/Core/Framework/DataAbstractionLayer/Event).
+
+## Other PHP events
+
+If the [DAL events](#DAL events) do not suffice for your use case, there are a few other events built into Shopware.
```

Comment:
When you write it like that it sounds like the DAL events are the best and first entrypoint in application changes. From my POV these are the last kind of events I'd like to try. They are my last resort as they lack the sales channel context as this is far away from business logic. I assume that the general case is to hook into existing business logic process and therefore their events. Technically these events are best as every change will pass them but for a beginner this is not something I want to look at first.

Reply:
I agree. Also the headline might be misleading then, sounds like "Just the other useless events".
Will come up with some other headline and change the phrases here, thanks!